### PR TITLE
[Tech-Debt][LB] Change log level from ERROR to WARN/INFO

### PR DESCRIPTION
**Pull request by udinobi**
- Change log level from ERROR to WARN/INFO for non-errors 4xx status code
- In order to avoid confusion, invert Downstream <--> Upstream names.
- Bump dependencies and plugins' versions (even though still on Play 2.7)

### CODE REVIEW

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala [hunk 1]
```diff
@@ -77,20 +77,20 @@ trait DownstreamConnector extends HttpErrorFunctions with Logging with UsingCirc
...
     statusCode match {
-      case BAD_GATEWAY => logErrorAndGenUpstreamResponse(BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
+      case BAD_GATEWAY => logErrorAndGenDownstreamResponse(BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
       case SERVICE_UNAVAILABLE =>
-        logErrorAndGenUpstreamResponse(SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
-      case GATEWAY_TIMEOUT => logErrorAndGenUpstreamResponse(GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
-      case _ => logErrorAndGenUpstreamResponse(statusCode, message, SERVER_ERROR)
+        logErrorAndGenDownstreamResponse(SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
+      case GATEWAY_TIMEOUT => logErrorAndGenDownstreamResponse(GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
+      case _ => logErrorAndGenDownstreamResponse(statusCode, message, SERVER_ERROR)
```
Comment (glidester) [inline app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: I think all this logic could be simplified and centralised inside a general "logAndResponse" method. Atm we are just passing the statusCode and message down the stack, the only thing that changes here is the log level and the response sent to PEGA. Could this not be controled in one "logAndResponse" method matching status code to log level and response?

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala [hunk 2]
```diff
@@ -106,13 +106,27 @@ trait DownstreamConnector extends HttpErrorFunctions with Logging with UsingCirc
...
+    logger.info(s"Error($statusCode) for $requestId. $message")
+    Future.successful(ErrorItem(baseError).invalidNec)
+  }
+
+  private def logWarningAndGenDownstreamResponse[T](statusCode: Int, message: String, baseError: BaseError)(
+    implicit requestId: RequestId
+  ): ResponseValidation[T] = {
+    logger.warn(s"Error($statusCode) for $requestId. $message")
+    Future.successful(ErrorItem(baseError).invalidNec)
+  }
```
Comment (glidester) [inline app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: Again, feels like unnecessary duplication here where all we are changing is the log level used.

**Review by glidester — COMMENTED** [review]
Think we could simplify the logging and response logic

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala (hunk 2 above)
Comment (udinobi) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: :+1:

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala (hunk 1 above)
Comment (udinobi) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: I would prefer to keep it this way. tbh I do not like to much to have a method with a big `match`. This way the logic between 4xx and 5xx is at least separated, and the 2 methods are still small.

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala [hunk 3]
```diff
@@ -77,20 +77,20 @@ trait UpstreamConnector extends HttpErrorFunctions with Logging with UsingCircui
...
     statusCode match {
-      case BAD_GATEWAY => logErrorAndGenDownstreamResponse(BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
+      case BAD_GATEWAY => logAndGenDownstreamResponse(error, BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
       case SERVICE_UNAVAILABLE =>
-        logErrorAndGenDownstreamResponse(SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
-      case GATEWAY_TIMEOUT => logErrorAndGenDownstreamResponse(GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
-      case _ => logErrorAndGenDownstreamResponse(statusCode, message, SERVER_ERROR)
+        logAndGenDownstreamResponse(error, SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
+      case GATEWAY_TIMEOUT => logAndGenDownstreamResponse(error, GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
+      case _ => logAndGenDownstreamResponse(error, statusCode, message, SERVER_ERROR)
```
Comment (glidester) [inline app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: NIT: These two methods could be combined as they do the same thing really. Maybe something like: 
```
private def handleUpstreamError[T](statusCode: Int, message: String)(
    implicit r: RequestId
  ): ResponseValidation[T] = {
    val logMessage = s"Error($statusCode) for $r. $message"
    
    statusCode match {
      case NOT_FOUND => notFound(message)
      case FORBIDDEN => logAndGenDownstreamResponse(warning, logMessage, BREATHING_SPACE_EXPIRED)
      case CONFLICT => logAndGenDownstreamResponse(info, logMessage, CONFLICTING_REQUEST)
      case BAD_GATEWAY => logAndGenDownstreamResponse(error, logMessage, UPSTREAM_BAD_GATEWAY)
      case SERVICE_UNAVAILABLE => logAndGenDownstreamResponse(error, logMessage, UPSTREAM_SERVICE_UNAVAILABLE)
      case GATEWAY_TIMEOUT => logAndGenDownstreamResponse(error, logMessage, UPSTREAM_TIMEOUT)
      case _ => logAndGenDownstreamResponse(error, logMessage, SERVER_ERROR)
    }
  }
```

**Review by glidester — APPROVED** [review]
Approved. With a suggestion ;)

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala (hunk 3 above)
Comment (udinobi) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala]: I got it :smiley_cat: But as I wrote, I like keeping separated 4xx and 5xx logic.

**Pull request merged**